WcaOnRails/app/models/competition.rb

```diff
@@ -863,7 +863,18 @@ def days_until
 
   def has_date_errors?
     valid?
-    !errors[:start_date].empty? || !errors[:end_date].empty? || (!showAtAll && days_until && days_until < SHOULD_BE_ANNOUNCED_GTE_THIS_MANY_DAYS)
+    !errors[:start_date].empty? || !errors[:end_date].empty? || (!showAtAll && days_until && days_until < MUST_BE_ANNOUNCED_GTE_THIS_MANY_DAYS)
+  end
+
+  # The competition must be at least 28 days in advance in order to confirm it. Admins are able to modify the competition despite being less than 28 days in advance.
+  validate :start_date_must_be_28_days_in_advance, if: :confirmed_or_visible?
+  def start_date_must_be_28_days_in_advance
+    if editing_user_id
+      editing_user = User.find(editing_user_id)
```

Comment:
This doesn't actually feel like the right way to use rails validations here: ideally all of our validations are written in a way such that they would pass on every row of every table in our database. This looks like it would cause all competitions to become invalid once we're within 28 days of them. This feels more like it should be a `can_confirm_competition?` permission in the users table. That would also solve the weirdness that @viroulep mentions about the `editing_user_id` field.

Unrelated, but is this code accurate with regards to timezones?